### ISSUE TITLE
Duplicate target paths warning after upgrade to v0.115.0

### ISSUE THREAD

**Issue by xeptore** [gohugoio/hugo#11187]
### What version of Hugo are you using (`hugo version`)?

```sh
$ hugo version
hugo v0.115.0+extended linux/amd64 BuildDate=unknown
```

### Does this issue reproduce with the latest release?
Yes
---
After upgrading from v0.114.1 to the latest version, I receive duplicate path warning for every content file every time I change, and save a file.

Steps to reproduce:
1. Clone/download the repo [xeptore/blog](https://github.com/xeptore/blog)
    ```sh
    curl -sSfL https://github.com/xeptore/blog/archive/main.tar.gz | tar -xzvf -
    ```
6. Install dependencies using `pnpm` (`pnpm install`)
7. Start server: `make run`
8. Edit, and save a random content file, e.g., `layouts/404.html`

If you follow the above steps having hugo version 0.114.1, no path warnings will be printed in the terminal. But,  if you upgrade the hugo version to v0.115.0, and try step 4, a duplicate target path warning will be printed similar to the following:
```txt
WARN  Duplicate target paths: /404.html (2), /about/index.html (2), /contact/index.html (2), /emoji-support-demo/index.html (2), /index.html (2), /index.xml (2), /markdown-syntax-guide/index.html (2), /math-typesetting-demo/index.html (2), /my-first-post-with-slug/index.html (2), /rich-content-demo/index.html (2), /sitemap.xml (2), /tags/demo/index.html (2), /tags/demo/index.xml (2), /tags/development/index.html (2), /tags/development/index.xml (2), /tags/emoji/index.html (2), /tags/emoji/index.xml (2), /tags/first/index.html (2), /tags/first/index.xml (2), /tags/hello/index.html (2), /tags/hello/index.xml (2), /tags/markdown/index.html (2), /tags/markdown/index.xml (2), /tags/privacy/index.html (2), /tags/privacy/index.xml (2), /tags/shortcodes/index.html (2), /tags/shortcodes/index.xml (2), pages/1/index.html (2), robots.txt (2), tags/demo/pages/1/index.html (2), tags/development/pages/1/index.html (2), tags/emoji/pages/1/index.html (2), tags/first/pages/1/index.html (2), tags/hello/pages/1/index.html (2), tags/markdown/pages/1/index.html (2), tags/privacy/pages/1/index.html (2), tags/shortcodes/pages/1/index.html (2)
```

**Comment by bep** [comment]
This is a side effect of the fix for fa0e16f4c79a703d122f1e3a3a99f4b779aea9b2 

We will fix this, but to me it does not make sense to have this enabled when running the server, so I suggest you at least temporarily turn it off in your server setup (or config) and just do `hugo --printPathWarnings` when you need to do these checks.

**Comment by xeptore** [comment]
@bep,  #11188 fixed this issue as I installed `master` version, and it no warnings were printed even in server mode.

Thanks for the effort and being quick to respond!
